[Feat] 카테고리 선택용 Chip 공통 컴포넌트 구현

seojeongm opened the pull request:
## 📌 Summary
- close #20 
- 카테고리 선택용 Chip 컴포넌트를 구현했습니다. 
- 선택 여부에 따라 배경/텍스트 색이 변경되며, 클릭 시 상태가 toggle됩니다. 
- 공통 컴포넌트로 분리하여 다양한 곳에서 재사용할 수 있도록 구성했습니다.

## 📝 Tasks
- Chip 컴포넌트 생성 (`src/shared/components/Chip/index.tsx`)
- `label`, `selected`, `onClick` props 정의
- 선택 상태에 따라 스타일 동적 적용
  - 선택됨: 파란 배경, 흰 텍스트
  - 선택되지 않음: 흰 배경, 검정 테두리, 검정 텍스트
- toggle 동작 구현
- ChipTest 테스트 페이지 추가

## ✅ PR Checklist (To Reviewer)
- [ ] 선택 여부에 따라 시각적 스타일이 올바르게 적용되었는지 확인
- [ ] 클릭 시 선택 상태가 정상적으로 toggle되는지 확인 
- [ ] 컴포넌트 구조가 재사용 가능하게 잘 분리되었는지 확인 

## 📸 Screenshot
https://github.com/user-attachments/assets/8d34e722-badc-4dc2-b840-0de78051877b


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Introduced a new component for displaying and selecting category chips, allowing users to choose multiple categories with visual feedback.
* **Style**
  * Improved consistency in code formatting and semicolon usage within configuration files.
* **Chores**
  * Enhanced linting rules for better code quality, accessibility, security, and formatting enforcement.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

seojeongm committed:
feat: Chip 컴포넌트 UI 구현 및 테스트 페이지 추가

seojeongm commented:
공통 css 적용 후 `absolute` 대신 반응형으로 코드 변경 필요.

maehwasoo commented on src/pages/ChipTest.tsx:
merge하기 전에 이 테스트 파일 삭제하고 커밋 한 번하면 될 것 같습니다~!

maehwasoo commented on src/shared/components/Chip/index.tsx:
```diff
@@ -0,0 +1,20 @@
+    onClick: () => void;
+}
+
+const Chip: React.FC<ChipProps> = ({ label, selected, onClick }) => {
```
React.FC는 현재 공식적으로 권장하지 않는 예전 패턴입니다

불필요한 상황에서 React.FC가 자동으로 children prop을 추가해버리기도 하고, 
TypeScript의 고급 기능들(제네릭 타입 지원 불가, defaultProps 등)과 충돌, React 19의 새로운 기능들(ref as prop, use API 등)과도 호환되지 않습니다

따라서 React.FC는 제거해주세요~!

seojeongm replied:
확인했습니다! React.FC를 사용하지 않도록 코드 수정하고, 머지 시 테스트 파일은 지우겠습니다. 자세히 봐주셔서 감사합니다! :)

maehwasoo commented on src/shared/components/Chip/index.tsx:
컴포넌트 파일명을 index.tsx로 하면 IDE에서 여러 index 구분이 어렵고 깃 추적도 어려워집니당
단일 컴포넌트의 경우에는 파일명을 명시적으로 정하는 것이 가독성 측면에서 훨씬 유리해 보입니다
Chip.tsx나 CategoryChip 등 파스칼 케이스의 명시적 컴포넌트명으로 하면 더욱 좋을 것 같습니다

seojeongm replied:
- `React.FC` 제거
- 컴포넌트 파일명 변경 (`index.tsx` -> `CategoryChips.tsx`)
- 테스트 파일 제거 

모두 완료했습니다 :)

maehwasoo reviewed (approved):
고생하셨습니다 ~ 리뷰만 한 번 확인해주세요 !

minjaehan reviewed (approved):
수정사항 반영한 후에 머지하기 전에 코드 충돌 처리만 하면 될 것 같습니다 고생하셨어요~

seojeongm committed:
Merge remote-tracking branch 'origin/develop' into feat/chip-component/#20

seojeongm committed:
refactor: rename CategoryChip 컴포넌트 파일명 index.tsx → CategoryChip.tsx

seojeongm committed:
refactor: CategoryChip 컴포넌트에서 React.FC 제거

seojeongm committed:
style: CategoryChip 컴포넌트에 커스텀 CSS 적용

seojeongm committed:
refactor: Chip 관련 파일 이름 및 구조 리팩토링

- ChipTest.tsx 삭제
- 기존 CategoryChip.tsx 및 Chip/index.tsx 제거
- 새로운 CategoryChips.tsx 파일로 통합 및 재작성
- 컴포넌트 구조 및 네이밍 정리
